doc/features/README.md

```diff
@@ -0,0 +1,64 @@
+Features introspection in e2e tests
+======================================
+
+**Goal**
+
+Provide a way to identify the active features for every supported operator version.
+
+**Design**
+
+- Provide the possibility to group a set of tests that verify a common feature using ginkgo Label("feature:<topic>").
+  A Topic should describe the main feature being tested or the unique effect of the test on the cluster. For instance:
+  `feature:schedrst`: indicates that the test is expected to restart the schedular, be it a pod restart or a complete removal of the scheduler and recreation.
+  `feature:rtetols`: indicates that the test is testing the new feature RTE tolerations, which involves specific non-default configuration on the operator CRs hence functional effects.
+  `feature:unsched`: indicates that the test workload is anticipated to be un-schedulable for any reason for example: insufficient resources (pod is pending), TAE (failed)..
+  `feature:wlplacement`: indicates that the test challenges the scheduler to place a workload on a node. This is a common label for most of the functional tests.
+    - Rules for creating a new topic
+        - A Topic should point out briefly the new feature being tested or the unique effect of the test on the cluster
+        - The topic should consist of small letters
+        - should not consist of any of `&|!,()/`
+
+- Define the list of supported features for each version: this is documented in a json format in pkg/features/_topics.json under the corresponding branch.
+  Adding tests for new feature requires updating the _topics.json file for each branch that its release supports that feature.
+  For instance, RTE tolerations are supported starting 4.14. So tests labeled with `Label("feature:rtetols")` should not run below 4.14.
+  Similarly, pausing refreshing the NRT info is supported starting 4.16, and such tests are associated with `Label("feature:rtetols")` should not run below 4.16.
+
+- Allow the user to list the supported features by adding new flag, `-inspect-features` which if passed the numaresources-operator binary on the controller pod will list the active features for the deployed version of the operator.
+
+- For automated bugs' scenarios that their fix is not back-ported to all versions:
+    - **TBD**
+
+**Deprecation**
+
+Tagging tests to a specific feature functionality were done by adding tags in the spec description like `It("[rtetols][distruptive]...")`. Starting Jul 2024, while the master branch is pointing to 4.17, the new preferred way to tag tests is using ginkgo `Label("")` such as `Context("should run pods requesting host-level resources", Label("hostlevel","distruptive"), func(){..})`. Tags aren't forbidden yet; they are just deprecated, so whenever a tag is added, a label is also required.
```

Comment:
this is true and correct, but doesn't belong to the design of the features inspection